lib/sshkit/ssh/connection.ex

```diff
@@ -42,27 +46,51 @@ defmodule SSHKit.SSH.Connection do
     open(to_charlist(host), options)
   end
   def open(host, options) do
-    port = Keyword.get(options, :port, 22)
-    timeout = Keyword.get(options, :timeout, :infinity)
-    ssh_modules = Keyword.get(options, :ssh_modules, @ssh_modules)
-    ssh = erlang_module(%{ssh_modules: ssh_modules}, :ssh)
+    {ssh_options, sshkit_options} = fetch_options(options)
+
+    ssh = ssh_module(sshkit_options)
+    case ssh.connect(host, sshkit_options.port, ssh_options, sshkit_options.timeout) do
+      {:ok, ref} -> {
+        :ok,
+        %Connection{
+          host: host,
+          port: sshkit_options.port,
+          options: ssh_options,
+          ref: ref,
+          ssh_modules: sshkit_options.ssh_modules
+        }
+      }
+      err -> err
+    end
+  end
 
-    defaults = [user_interaction: false]
+  @default_erlang_ssh_options [user_interaction: false]
+  defp fetch_options(options) do
+    dry_run = Keyword.get(options, :dry_run, false)
+    ssh_modules = if dry_run do
+      @dry_run_ssh_modules
+    else
+      @ssh_modules
+    end
 
-    options =
-      defaults
+    sshkit_options = %{
+      port: Keyword.get(options, :port, 22),
+      timeout: Keyword.get(options, :timeout, :infinity),
+      ssh_modules: Keyword.get(options, :ssh_modules, ssh_modules),
+      dry_run: dry_run
+    }
+
+    erlang_ssh_options =
+      @default_erlang_ssh_options
       |> Keyword.merge(options)
-      |> Keyword.drop([:port, :timeout, :ssh_modules])
+      |> Keyword.drop([:port, :timeout, :ssh_modules, :dry_run])
```

Comment:
Now that this list is growing, maybe it would also make sense to extract it out into a module var, e.g. `sshkit_connection_options` or whatever would make sense to express the fact that these options are not compatible with the underlying Erlang modules.